hr_employee_service/models/hr_employee.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+from dateutil.relativedelta import relativedelta
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    service_hire_date = fields.Date(
+        string='Hire Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Hire date is normally the date an employee completes new hire'
+            ' paperwork'
+        ),
+    )
+    service_start_date = fields.Date(
+        string='Start Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Start date is the first day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_termination_date = fields.Date(
```

Comment:
Aren' these a duplication on the Contract start and end dates?
At least the service start duration should hire date, if no service start date is provided.

Reply:
Not quite, since Contracts may be expired, renewed, prolonged etc., thus not a reliable source of information. Difference between hire and start date is as well significant - hire date is when paperwork was actually finished (offer accepted and signed e.g.) while start date may not yet be defined, thus assuming if there's no start date yet there's hire date then hire is start - is a bit long-shot.

Reply:
Yes contracts can be renewed, and contract start date is not the same as hire date.
But the date should be the first day of work, and the start date of the first contract.
If you need e date for when the paperwork was finished, that should be a different field.

Reply:
First contract date - that would work except for the re-hire cases

Reply:
Regrading date when paperwork is done - that's commonly referred to as a hire date, so not sure if I got you correctly. (I did some googling on the matter)